king_phisher/server/database/models.py

```diff
@@ -397,6 +398,95 @@ class Message(Base):
 	credentials = sqlalchemy.orm.relationship('Credential', backref='message', cascade='all, delete-orphan')
 	visits = sqlalchemy.orm.relationship('Visit', backref='message', cascade='all, delete-orphan')
 
+@register_table
+class ModuleQuestion(Base):
+	__repr_attributes__ = ('question',)
+	__tablename__ = 'module_questions'
+	id = sqlalchemy.Column(sqlalchemy.Integer, primary_key=True)
+	question = sqlalchemy.Column(sqlalchemy.String, nullable=False)
+	question_type = sqlalchemy.Column(sqlalchemy.String, nullable=False)
+	hint = sqlalchemy.Column(sqlalchemy.String)
+	url_reference = sqlalchemy.Column(sqlalchemy.String)
+	possible_answer_group_id = sqlalchemy.Column(sqlalchemy.Integer, nullable=False)
+	question_answer_id = sqlalchemy.Column(sqlalchemy.Integer, sqlalchemy.ForeignKey('question_answers.id'))
+	# relationships
+	submission_answer_link = sqlalchemy.orm.relationship('SubmissionAnswerLink', backref='module_questions', cascade='all, delete-orphan')
+
+@register_table
+class Module(TagMixIn, Base):
+	__tablename__ = 'modules'
+	created = sqlalchemy.Column(sqlalchemy.DateTime, nullable=False)
+	# relationships
+	module_question = sqlalchemy.orm.relationship('ModuleQuestion', backref='modules', cascade='all, delete-orphan')
+	test_module = sqlalchemy.orm.relationship('TestModule', backref='modules', cascade='all, delete-orphan')
+
+@register_table
+class PossibleAnswerLink(Base):
+	__repr_attributes__ = ('group_id', 'answer_id')
+	__tablename__ = 'possible_answer_link'
+	id = sqlalchemy.Column(sqlalchemy.Integer, primary_key=True)
+	question_answer_id = sqlalchemy.Column(sqlalchemy.Integer, sqlalchemy.ForeignKey('question_answers.id'), nullable=False)
+	group_id = sqlalchemy.Column(sqlalchemy.Integer)
+
+@register_table
+class QuestionAnswer(Base):
+	__repr_attributes__ = ('answer',)
+	__tablename__ = 'question_answers'
+	id = sqlalchemy.Column(sqlalchemy.Integer, primary_key=True)
+	answer = sqlalchemy.Column(sqlalchemy.String, nullable=False, unique=True)
```

Comment:
Unique should probably be false since the intention of `case_sensitive` changes it's interpretation. We'll just have to enforce it's uniqueness in code.